src/main/java/com/twentyonec/ItemsLogger/utils/Storage.java

```diff
@@ -0,0 +1,261 @@
+package com.twentyonec.ItemsLogger.utils;
+
+import java.sql.Connection;
+import java.sql.Date;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Time;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+
+import com.twentyonec.ItemsLogger.ItemPlayer;
+import com.twentyonec.ItemsLogger.ItemsLogger;
+import com.zaxxer.hikari.HikariConfig;
+import com.zaxxer.hikari.HikariDataSource;
+
+public class Storage {
+
+	private static Storage storage = null;
+
+	private ItemsLogger plugin = ItemsLogger.getPlugin();
+
+	private String hostname;
+	private String port;
+	private String username;
+	private String password;
+	private String database;
+
+	private HikariDataSource dataSource;
+
+	private Storage(final String hostname, final String port, final String username,
+			final String password, final String database) {
+
+		this.hostname = hostname;
+		this.port = port;
+		this.username = username;
+		this.password = password;
+		this.database = database;
+
+	}
+
+	public static synchronized Storage getStorage(ItemsLogger plugin) {
+		final Config config = plugin.getConfigManager();
+
+		if (storage == null) {
+			storage = new Storage(config.getHostname(), config.getPort(), config.getUsername(),
+					config.getPassword(), config.getDatabase());
+		}
+
+		return storage;
+	}
+
+	private void connect() {
+
+		final String jdbcUrl = "jdbc:mysql://" + this.hostname + ":" + this.port + "/" + this.database;
+
+		HikariConfig config = new HikariConfig();
+
+		config.setPoolName("itemslogger");
+		config.setJdbcUrl(jdbcUrl);
+		config.setUsername(this.username);
+		config.setPassword(this.password);
+
+		config.addDataSourceProperty("cachePrepStmts", "true");
+		config.addDataSourceProperty("prepStmtCacheSize", "250");
+		config.addDataSourceProperty("prepStmtCacheSqlLimit", "2048");
+		config.addDataSourceProperty("useServerPrepStmts", "true");
+		config.addDataSourceProperty("rewriteBatchedStatements", "true");
+		config.addDataSourceProperty("maintainTimeStats", "false");
+
+		config.setMaximumPoolSize(10);
+		config.setMinimumIdle(10);
+		config.setIdleTimeout(300000);
+		config.setMaxLifetime(600000);
+		config.setConnectionTimeout(5000);
+		config.setInitializationFailTimeout(-1);
+
+		try {
+			this.dataSource = new HikariDataSource(config);
+		} catch (Exception e) {
+			e.printStackTrace();
+		}
+
+	}
+
+	public Connection getConnection() throws SQLException {
```

Comment:
This shouldn't be public if all database updates come through this class
```suggestion
	private Connection getConnection() throws SQLException {
```